ting-bill-split.go

```diff
@@ -362,35 +364,166 @@ func createBillsFile(path string) {
+	var msgFile *os.File
+	var megFile *os.File
 
-	billPtr := flag.String("bills", "", "filename for bills toml - ex: -bills=\"bills.toml\"")
-	minPtr := flag.String("minutes", "", "filename for minutes csv - ex: -minutes=\"minutes.csv\"")
-	msgPtr := flag.String("messages", "", "filename for messages csv - ex: -messages=\"messages.csv\"")
-	megPtr := flag.String("megabytes", "", "filename for megabytes csv - ex: -megabytes=\"megabytes.csv\"")
+	if err != nil {
+		log.Fatal(err)
+	}
 
-	badParam := false
-	paramMap := map[string]*string{
-		"bills":     billPtr,
-		"minutes":   minPtr,
-		"messages":  msgPtr,
-		"megabytes": megPtr,
+	for _, file := range files {
+		if !file.IsDir() {
+			fmt.Println(file.Name())
+
+			if billFile == nil && isFileMatch(file.Name(), "bills", "toml") {
+				billFile, err = os.Open(file.Name())
+				if err != nil {
+					log.Fatal(err)
+				}
+				fmt.Println(billFile)
+			}
+
+			if minFile == nil && isFileMatch(file.Name(), "minutes", "csv") {
+				minFile, err = os.Open(file.Name())
+				if err != nil {
+					log.Fatal(err)
+				}
+				fmt.Println(minFile)
+			}
+
+			if msgFile == nil && isFileMatch(file.Name(), "messages", "csv") {
+				msgFile, err = os.Open(file.Name())
+				if err != nil {
+					log.Fatal(err)
+				}
+				fmt.Println(msgFile)
+			}
+
+			if megFile == nil && isFileMatch(file.Name(), "megabytes", "csv") {
+				megFile, err = os.Open(file.Name())
+				if err != nil {
+					log.Fatal(err)
+				}
+				fmt.Println(megFile)
+			}
+		}
 	}
 
+	if billFile == nil || minFile == nil || msgFile == nil || megFile == nil {
+		fmt.Println("Unable to open necessary files.")
+
+		if billFile == nil {
+			fmt.Println("Bills file not found.")
+		}
+
+		if minFile == nil {
+			fmt.Println("Minutes file not found.")
+		}
+
+		if msgFile == nil {
+			fmt.Println("Messages file not found.")
+		}
+
+		if megFile == nil {
+			fmt.Println("Megabytes file not found.")
+		}
+	} else {
+		fmt.Printf("\nRunning calculations based on files in directory: %s\n\n", path)
+
+		billData, err := parseBill(billFile)
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		// TODO remove this once we do something useful with it.
+		fmt.Println("billData ... something something ... *wanders off*")
+		fmt.Println(billData)
+
+		minMap, err := parseMinutes(minFile)
+		if err != nil {
+			log.Fatal(err)
+		}
+		fmt.Println(minMap)
+
+		msgMap, err := parseMessages(msgFile)
+		if err != nil {
+			log.Fatal(err)
+		}
+		fmt.Println(msgMap)
+
+		megMap, err := parseMegabytes(megFile)
+		if err != nil {
+			log.Fatal(err)
+		}
+		fmt.Println(megMap)
+
+		//TODO take in each map and return a billSplit
+		split, err := parseMaps(minMap, msgMap, megMap, billData)
+		if err != nil {
+			log.Fatal(err)
+		}
+		fmt.Println(split)
+	}
+}
+
+func main() {
+	fmt.Printf("Ting Bill Splitter\n\n")
+
 	flag.Parse()
 	args := flag.Args()
 
 	if len(args) > 0 {
 		fmt.Println("Running in batch mode")
-		if args[0] == "new" {
+
+		command := args[0]
```

Comment:
Would be a good place to use `switch args[0]` instead of a bunch of if/elseif/else

Reply:
Why did I never think to look whether there was a `switch` for go?  Python brain, I guess.